src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,138 @@
+package duke;
+
+import java.util.ArrayList;
+
+import duke.exception.DukeException;
+import duke.exception.InvalidInput;
+import duke.task.Task;
+
+
+/**
+ * A class that encapsulates the storing of all the task data.
+ */
+public class TaskList {
+    /** The ArrayList that contains all the tasks */
+    private final ArrayList<Task> data;
+
+    /**
+     * The class constructor that initialises the ArrayList.
+     */
+    public TaskList() {
+        this.data = new ArrayList<>();
+    }
+
+    /**
+     * Returns the number of tasks saved.
+     *
+     * @return The size of the ArrayList.
+     */
+    public int size() {
+        return data.size();
+    }
+
+    /**
+     * Returns the tasks saved in the provided index.
+     *
+     * @param index The index of the tasks of interest.
+     * @return The task of interest.
+     */
+    public Task get(int index) {
+        return data.get(index);
+    }
+
+    /**
+     * Appends a new task to the ArrayList.
+     *
+     * @param task The new task to append.
+     */
+    public void add(Task task) {
+        data.add(task);
+    }
+
+    /**
+     * Checks if the number provided is valid and returns the number in int.
+     *
+     * @param res The string to check.
+     * @return The valid index in Integer.
+     * @throws InvalidInput If the string given is not a number or the number is
+     *                      less than 0 or more than the size of the ArrayList.
+     */
+    public int stringIndexToInt(String res) throws InvalidInput {
```

Comment:
Perhaps prepending this method with an appropriate verb (e.g. 'get' or 'convert') would be better?